crepecake/src/main/java/com/google/cloud/tools/crepecake/cache/CacheMetadata.java

```diff
@@ -31,7 +31,7 @@
 
   private final ImageLayers<TimestampedCachedLayer> baseImageLayers = new ImageLayers<>();
 
-  private Map<ApplicationLayerType, TimestampedCachedLayer> applicationLayers = new HashMap<>();
+  private Map<CachedLayerType, TimestampedCachedLayer> applicationLayers = new HashMap<>();
```

Comment:
There can be only one layer per type?

Reply:
This class will be changed to match the new format. I added a TODO note. The change will be added in a followup PR that includes the translator to/from the cache metadata and its JSON.

Reply:
Why is BASE a CachedLayerType and also stored in baseImageLayers?

Reply:
This class is completely rewritten in #71 and will not have two separate fields for base layers and application layers.

Reply:
In that case, should we just merge this upstream and get rid of this PR?